src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -269,6 +269,22 @@ private bool TreatAllCodeAsNonGeneratedCode
             }
         }
 
+        private ConcurrentSet<string>? _lazySuppressedDiagnosticIdsForUnsuppressedAnalyzers;
+
+        /// <summary>
+        /// Lazily populated set of diagnostic IDs which are suppressed for some part of the compilation (tree/folder/entire compilation),
+        /// but the analyzer reporting the diagnostic is itself not suppressed for the entire compilation, i.e. the analyzer
+        /// belongs to <see cref="UnsuppressedAnalyzers"/>.
+        /// </summary>
+        private ConcurrentSet<string> SuppressedDiagnosticIdsForUnsuppressedAnalyzers
+        {
+            get
+            {
+                Debug.Assert(_lazySuppressedDiagnosticIdsForUnsuppressedAnalyzers != null);
```

Comment:
>_lazySuppressedDiagnosticIdsForUnsuppressedAnalyzers != null [](http://example.com/codeflow?start=29&length=60)

How do we know `_lazySuppressedDiagnosticIdsForUnsuppressedAnalyzers != null`? It looks like the field is only set by a parameter to `Initialize()` which is marked nullable.

Reply:
This is the pattern used for most lazy backing fields and property pairs in this file. The property getter is asserting that it is accessed only on those code paths where this field is guaranteed to be non-null. In this case, the API `GetAllDescriptors` added in this PR is invoked from the command line compiler only when `/errorlog` switch is specified on the command line, which it turns passes `trackSuppressedDiagnosticIds: true` to the analyzer driver's `Initialize()` method and ensures the field is assigned a non-null value during initialization.